src/js/dynload.ts

```diff
@@ -110,4 +140,84 @@ export async function loadDynlib(lib: string, global: boolean) {
   }
 }
 
+/**
+ * Load dynamic libraries inside a package.
+ *
+ * This function handles some painful details of loading dynamic libraries:
+ * - We need to load libraries in the correct order considering dependencies.
+ * - We need to load libraries globally if they are required by other libraries.
+ * - We need to tell Emscripten where to search for libraries.
+ * - The dynlib metadata inside a wasm module only contains the library name, not the path.
+ *   So we need to handle them carefully to avoid loading the same library twice.
+ *
+ * @param pkg The package metadata
+ * @param dynlibPaths The list of dynamic libraries inside a package
+ * @private
+ */
+export async function loadDynlibsFromPackage(
+  pkg: PackageData,
+  dynlibPaths: string[],
+) {
+  // assume that shared libraries of a package are located in <package-name>.libs directory,
+  // following the convention of auditwheel.
+  const auditWheelLibDir = `${API.sitepackages}/${
+    pkg.file_name.split("-")[0]
+  }.libs`;
+
+  // This prevents from reading large libraries multiple times.
+  const readFileMemoized: ReadFileType = memoize(Module.FS.readFile);
+
+  const globalLibs: Set<string> = calculateGlobalLibs(
+    dynlibPaths,
+    readFileMemoized,
+  );
+
+  const dynlibs = dynlibPaths.map((path) => {
+    const global = globalLibs.has(Module.PATH.basename(path));
+    return {
+      path: path,
+      global: global || !!pkg.shared_library,
+    };
+  });
+
+  // Sort libraries so that global libraries can be loaded first.
+  // TODO(ryanking13): It is not clear why global libraries should be loaded first.
+  //    But without this, we get a fatal error when loading the libraries.
+  dynlibs.sort((lib: { global: boolean }) => (lib.global ? -1 : 1));
```

Comment:
This is a bit wasteful I think, better would be:
```suggestion
type T = { global: boolean };
dynlibs.sort((lib1: T, lib2 : T) => (lib2.global - lib1.global));
```